lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,28 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.read(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            s = input.readpartial(4096)
```

Comment:
ooooh `rpartition` looks useful, but I'll definitely need a loop there. It'll also allow me to match `\r`!